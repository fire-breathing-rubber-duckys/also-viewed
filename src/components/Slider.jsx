/* eslint-disable no-confusing-arrow */
import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const FlexBox = styled.div`
  display: flex;
  overflow: contained;
`;

const Page = styled.div`
  flex: 0 0 auto;
  opacity: ${(props) => props.active ? 1 : 0};
  transition: all 500ms ease;
  width: 100%;
`;

const Pages = styled.div`
  display: flex;
  ${(props) => props.currentPage
    && css` transform: translateX(-${props.currentPage * 100}%);`};
  transition: all 700ms ease;
`;

const Slider = ({ children, currentPage }) => {
  const activePage = children.map((page, i) => (
    <Page active={currentPage === i} key={i}>
      {page}
    </Page>
  ));

  return (
    <div>
      <FlexBox className="flex-box">
        <Pages currentPage={currentPage} className="pages">
          {activePage}
        </Pages>
      </FlexBox>
    </div>
  );
};

Slider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  currentPage: PropTypes.number.isRequired,
};

export default Slider;
