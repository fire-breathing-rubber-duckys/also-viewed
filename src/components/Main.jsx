import React, { useState, useEffect } from 'react';
import axios from 'axios';
import styled from 'styled-components';
import Slider from './Slider';
import Page from './Page';

const LeftButton = styled.div`
  position: relative;
  left: 10px;
  top: 170px;
  z-index: 3;
`;

const RightButton = styled.div`
  position: relative;
  left: 1120px;
  bottom: 200px;
  z-index: 3;
`;

const ContainerView = styled.div`
  position: relative;
  top: -20px;
`;

const ContainerBought = styled.div`
  position: relative;
  top: -80px;
`;

const Main = () => {
  const [currentPageOne, setCurrentPageOne] = useState(0);
  const [currentPageTwo, setCurrentPageTwo] = useState(0);
  const [productName, setProductName] = useState('');
  const [pageOneData, setPageOneData] = useState([]);
  const [pageTwoData, setPageTwoData] = useState([]);

  useEffect(() => {
    axios.get('http://localhost:1337/api/top-picks/fetch')
      .then((response) => {
        setPageOneData(response.data);
      })
      .catch((error) => {
        throw error;
      })
      .then(() => {
        axios.get('http://localhost:1337/api/ultimately-bought/fetch')
          .then((response) => {
            setPageTwoData(response.data[0].related);
            setProductName(` ${response.data[0].productName} `);
          })
          .catch((error) => {
            throw error;
          });
      });
  }, []);

  let viewBtnLeft;
  let viewBtnRight;
  let boughtBtnLeft;
  let boughtBtnRight;

  if (currentPageOne === 0) {
    viewBtnLeft = 'hidden-view';
    viewBtnRight = 'button-view';
  } else {
    viewBtnLeft = 'button-view';
    viewBtnRight = 'hidden-view';
  }

  if (currentPageTwo === 0) {
    boughtBtnLeft = 'hidden-bought';
    boughtBtnRight = 'button-bought';
  } else {
    boughtBtnLeft = 'button-bought';
    boughtBtnRight = 'hidden-bought';
  }

  return (
    <div className="main-view">
      <ContainerView className="container-viewed">
        <h1 className="title"> People Also Viewed </h1>
        <LeftButton>
          <button
            type="button"
            className={viewBtnLeft}
            onClick={() => { setCurrentPageOne(0); }}
          >
            <i className="fa fa-long-arrow-left" />
          </button>
        </LeftButton>

        <Slider currentPage={currentPageOne} className="slider">
          <Page data={pageOneData.slice(0, 6)} page={currentPageOne} />
          <Page data={pageOneData.slice(5, pageOneData.length)} page={currentPageOne} />
        </Slider>

        <RightButton>
          <button
            type="button"
            className={viewBtnRight}
            onClick={() => { setCurrentPageOne(1); }}
          >
            <i className="fa fa-long-arrow-right" />
          </button>
        </RightButton>
      </ContainerView>

      <ContainerBought className="container-bought">
        <h1 className="title">
          People shopping
          {productName}
          ultimately bought
        </h1>
        <LeftButton>
          <button
            type="button"
            className={boughtBtnLeft}
            onClick={() => { setCurrentPageTwo(0); }}
          >
            <i className="fa fa-long-arrow-left" />
          </button>
        </LeftButton>

        <Slider currentPage={currentPageTwo} className="slider">
          <Page data={pageTwoData.slice(0, 6)} page={currentPageTwo} />
          <Page data={pageTwoData.slice(5, pageTwoData.length)} page={currentPageTwo} />
        </Slider>

        <RightButton>
          <button
            type="button"
            className={boughtBtnRight}
            onClick={() => { setCurrentPageTwo(1); }}
          >
            <i className="fa fa-long-arrow-right" />
          </button>
        </RightButton>
      </ContainerBought>
    </div>
  );
};

export default Main;
