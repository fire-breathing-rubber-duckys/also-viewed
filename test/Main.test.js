import React from 'react';
import { shallow } from 'enzyme';
import { expect as expectChai } from 'chai';
import renderer from 'react-test-renderer';
import Main from '../src/components/Main';
import Slider from '../src/components/Slider';

it('Should render correctly', () => {
  const tree = renderer
    .create(<Main />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Should render the correct content', () => {
  const wrapper = shallow(<Main />);
  it('Should render the viewed-container', () => {
    expectChai(wrapper.find('.container-viewed')).to.be.lengthOf(1);
  });
  it('Should render the bought-container', () => {
    expectChai(wrapper.find('.container-bought')).to.be.lengthOf(1);
  });
  it('Should render two sliders', () => {
    expectChai(wrapper.find(Slider)).to.be.lengthOf(2);
  });
  it('Should render one button-view button on load', () => {
    expectChai(wrapper.find('.button-view')).to.be.lengthOf(1);
  });
  it('Should render one hidden-view button on load', () => {
    expectChai(wrapper.find('.hidden-view')).to.be.lengthOf(1);
  });
  it('Should render one button-bought button on load', () => {
    expectChai(wrapper.find('.button-bought')).to.be.lengthOf(1);
  });
  it('Should render one hidden-bought button on load', () => {
    expectChai(wrapper.find('.hidden-bought')).to.be.lengthOf(1);
  });
});


describe('User interface', () => {
  const wrapper = shallow(<Main />);
  it('Should render one view button when one view button is clicked', () => {
    wrapper.find('.button-view').simulate('click');
    expectChai(wrapper.find('.button-view')).to.be.lengthOf(1);
  });
  it('Should render one bought button when one view button is clicked', () => {
    wrapper.find('.button-bought').simulate('click');
    expectChai(wrapper.find('.button-bought')).to.be.lengthOf(1);
  });
});