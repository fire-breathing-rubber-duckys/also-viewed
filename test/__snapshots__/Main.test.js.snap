// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="main-view"
>
  <div
    className="sc-iBPRYJ KIsfl container-viewed"
  >
    <h1
      className="title"
    >
       People Also Viewed 
    </h1>
    <div
      className="sc-jSgupP gRniap"
    >
      <button
        className="hidden-view"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-long-arrow-left"
        />
      </button>
    </div>
    <div>
      <div
        className="sc-bdfBwQ eHfRaG flex-box"
      >
        <div
          className="sc-dlfnbm hTUCeX pages"
        >
          <div
            className="sc-gsTCUz ePRiRs"
          >
            <div
              className="sc-eCssSg ifsWGG"
            />
          </div>
          <div
            className="sc-gsTCUz koMqhJ"
          >
            <div
              className="sc-eCssSg ifsWGG"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-gKsewC jzJMhD"
    >
      <button
        className="button-view"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-long-arrow-right"
        />
      </button>
    </div>
  </div>
  <div
    className="sc-fubCfw iyZZIK container-bought"
  >
    <h1
      className="title"
    >
      People shopping
      
      ultimately bought
    </h1>
    <div
      className="sc-jSgupP gRniap"
    >
      <button
        className="hidden-bought"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-long-arrow-left"
        />
      </button>
    </div>
    <div>
      <div
        className="sc-bdfBwQ eHfRaG flex-box"
      >
        <div
          className="sc-dlfnbm hTUCeX pages"
        >
          <div
            className="sc-gsTCUz ePRiRs"
          >
            <div
              className="sc-eCssSg ifsWGG"
            />
          </div>
          <div
            className="sc-gsTCUz koMqhJ"
          >
            <div
              className="sc-eCssSg ifsWGG"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-gKsewC jzJMhD"
    >
      <button
        className="button-bought"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-long-arrow-right"
        />
      </button>
    </div>
  </div>
</div>
`;
